### ISSUE TITLE
Time Picker has screen reader issues on web

### ISSUE THREAD

**Issue by rami-a** [flutter/flutter#66062]
After reviewing the time picker on web with a screen reader, a few issues were noticed:

- For the hour/minute controls, I hear "min 0, max 2" even though the number, e.g. an hour, is set to "7" (ChromeOS only)
- When selecting "AM" vs "PM", they are neither reported as "buttons" or as "selected" to reflect their selection state (both macOS and ChromeOS) - The [time picker code](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/time_picker.dart#L561) currently marks these as "selected" but not as "buttons" so part of this is expected and a bug.
  - Note: I also noticed a lack of "selected" being spoken for other widgets such as `ToggleButtons`
- In "text input mode", text fields do not have accessibility labels

#Steps to Reproduce

1. Run the flutter gallery (or any app with the time picker) on web (both macOS and chromeOS were tested)
2. Enable screen reader
3. Observe that certain time picker elements are not read out correctly (see above for details)

**Expected results:** Correct screen reader experience

**Actual results:**  Partial screen reader experience

```
[✓] Flutter (Channel master, 1.22.0-2.0.pre.3, on Mac OS X 10.15.5 19F101, locale en-US)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
[✓] Chrome - develop for the web
[✓] Android Studio (version 4.1)
[✓] Connected device (4 available)
```

**Comment by nturgut** [comment]
I'll have a look into this. This might also be due to gallery code. On a few occasions, we needed fix flutter/gallery instead of the web engine code. I'll update the issue.

**Comment by nturgut** [comment]
I tested this feature, as indicated the time is announced wrong after 2. (for example for 4 screenreader output is 3). The buttons and navigation worked correctly. 

Test environment: MacOS, VoiceOver, Chrome

**Comment by rami-a** [comment]
@nturgut did the am/pm buttons correctly output which one is selected because when I tested they didn't

**Comment by rami-a** [comment]
also cc @vick08 who performed the a11y review and found these issues.

**Comment by rami-a** [comment]
The "selected" issue is related to https://github.com/flutter/flutter/issues/66673
